packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -58,21 +61,31 @@ export class Publisher {
       httpRequestTimeout,
       httpClient,
       disable,
+      oauthSettings: oauthSettings,
     }: PublisherProps,
     emitter: NodeEmitter
   ) {
     this._emitter = emitter
     this._maxRetries = maxRetries
     this._maxEventsInBatch = Math.max(maxEventsInBatch, 1)
     this._flushInterval = flushInterval
-    this._auth = b64encode(`${writeKey}:`)
     this._url = tryCreateFormattedUrl(
       host ?? 'https://api.segment.io',
       path ?? '/v1/batch'
     )
     this._httpRequestTimeout = httpRequestTimeout ?? 10000
     this._disable = Boolean(disable)
     this._httpClient = httpClient
+    this._writeKey = writeKey
+
+    if (oauthSettings) {
+      const TokenManager = dependencyInjection.get('TokenManager')
```

Comment:
The dependency injection itself is so that we can pass different TokenManager implementations depending on what runtime we're in.

Take a look at `index.workerd.ts` for example - thanks to the export mappings in package.json, `node.js` will load `index.ts`, which includes our real `TokenManager` constructor.

We don't have all the things required to get JWT signing working in cloudflare workers/vercel edge/service workers, so the export mappings tell all of the builds tools for those to load `index.workerd.ts` instead, which uses a NoOp TokenManager.

If we try to be clever with conditional imports instead, with runtime detection, the bundlers for these runtimes break because they aren't all smart enough to parse our conditions and will attempt to bundle things they don't support.

Reply:
Oh, I see now. It's to make it more of a per platform thing, than a user configurable one. Makes sense.